### PR TITLE
docs: overall improvements for 10.2

### PR DESCRIPTION
**Pull request by jrudolph**
A first small set of documentation improvements for 10.2

### CODE REVIEW

File: docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala [hunk 1]
```diff
@@ -210,85 +211,94 @@ class HttpServerExampleSpec extends AnyWordSpec with Matchers
...
     def myAuthenticator: Authenticator[User] = ???
-    def retrieveOrdersFromDB: Seq[Order] = ???
+    def retrieveOrdersFromDB: Future[Seq[Order]] = ???
     def myDbActor: ActorRef = ???
     def processOrderRequest(id: Int, complete: Order => Unit): Unit = ???
 
-    val route = concat(
-      path("orders") {
-        authenticateBasic(realm = "admin area", myAuthenticator) { user =>
+    lazy val binding = Http().newServerAt("localhost", 8080).bind(topLevelRoute)
```
Comment (johanandren) [inline docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala:218]: This new bind API is super neat 👍👍👍

File: docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala [hunk 2]
```diff
@@ -210,85 +211,94 @@ class HttpServerExampleSpec extends AnyWordSpec with Matchers
...
+        path("orders")(ordersRoute),
+        // extract URI path element as Int
+        pathPrefix("order" / IntNumber)(orderRoute),
+        pathPrefix("documentation")(documentationRoute),
+        path("oldApi" / Remaining) { pathRest =>
+          redirect("http://oldapi.example.com/" + pathRest, MovedPermanently)
+        }
+      )
+
+    lazy val ordersRoute: Route =
```
Comment (johanandren) [inline docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala:233]: perhaps noted elsewhere, so sorry if I missed that, but perhaps put add a comment that the different subroute trees can be defined in different classes as well

File: docs/src/test/scala/docs/http/scaladsl/SprayJsonExample.scala [hunk 3]
```diff
@@ -68,7 +67,7 @@ object SprayJsonExample2 {
           path("create-order") {
             entity(as[Order]) { order =>
               val saved: Future[Done] = saveOrder(order)
-              onComplete(saved) { done =>
+              onSuccess(saved) { _ => // we are not interested in the result value `Done` but only in the fact that it was suce
```
Comment (johanandren) [inline docs/src/test/scala/docs/http/scaladsl/SprayJsonExample.scala:70]: successful

Comment (jrudolph) [reply docs/src/test/scala/docs/http/scaladsl/SprayJsonExample.scala:70]:
```suggestion
              onSuccess(saved) { _ => // we are not interested in the result value `Done` but only in the fact that it was successful
```

File: docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala (hunk 2 above)
Comment (jrudolph) [reply docs/src/test/scala/docs/http/scaladsl/HttpServerExampleSpec.scala:233]:
```suggestion
    // For bigger routes, these sub-routes can be moved to separate files
    lazy val ordersRoute: Route =
```

File: docs/src/main/paradox/introduction.md [hunk 4]
```diff
@@ -23,7 +23,8 @@ However, if your application is not primarily a web application because its core
 
 Akka HTTP was designed specifically as “not-a-framework”, not because we don’t like frameworks, but for use cases where a framework is not the right choice. Akka HTTP is made for building integration layers based on HTTP and as such tries to “stay on the sidelines”. Therefore you normally don’t build your application “on top of” Akka HTTP, but you build your application on top of whatever makes sense and use Akka HTTP merely for the HTTP integration needs.
 
-On the other hand, if you prefer to build your applications with the guidance of a framework, you should give [Play Framework](https://www.playframework.com/) or [Lagom](https://www.lagomframework.com/) a try, which both use Akka internally.
+On the other hand, if you prefer to build your applications with the guidance of a framework, you should give [Play Framework](https://www.playframework.com/) or [Lagom](https://www.lagomframework.com/) a try, which both use Akka internally. If you
+come from Play and want to try Akka HTTP, we collected a @ref[side-by-side comparison](routing-dsl/play-comparison.md) of how some routing feature of Play map to Akka HTTP features.
```
Comment (ennru) [inline docs/src/main/paradox/introduction.md:27]:
```suggestion
come from Play and want to try Akka HTTP, we collected a @ref[side-by-side comparison](routing-dsl/play-comparison.md) to show how some Play routing features map to the Akka HTTP routing DSL.
```

**Pull request merged**